### ISSUE TITLE
[Bug]: mo panic with runtime error: index out of range [11] with length 4 when run tpcc benchmark test

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#6280]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):1feeb04254c6fdab0ae0061ecc32349136d6ee10
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

2022/11/04 17:15:52.841165 +0800 INFO frontend/util.go:474 query trace status {"connection_id": 4294967295, "statement": "commit", "status": "success"}
2022/11/04 17:15:52.841246 +0800 INFO frontend/mysql_cmd_executor.go:3202 query trace {"connection_id": 1021, "query": "execute __mo_stmt_id_2 using @__mo_stmt_var_0,@__mo_stmt_var_1", "vars": "2 , 4"}
2022/11/04 17:15:52.841883 +0800 INFO frontend/mysql_cmd_executor.go:3202 query trace {"connection_id": 1019, "query": "execute __mo_stmt_id_16 using @__mo_stmt_var_0,@__mo_stmt_var_1,@__mo_stmt_var_2", "vars": "2934.05 , 3 , 10"}
2022/11/04 17:15:52.842142 +0800 INFO frontend/util.go:474 query trace status {"connection_id": 4294967295, "statement": "start transaction", "status": "success"}
2022/11/04 17:15:52.842152 +0800 INFO frontend/mysql_cmd_executor.go:3202 query trace {"connection_id": 1018, "query": "execute __mo_stmt_id_16 using @__mo_stmt_var_0,@__mo_stmt_var_1,@__mo_stmt_var_2", "vars": "2411.53 , 10 , 1"}
2022/11/04 17:15:52.842171 +0800 INFO frontend/util.go:474 query trace status {"connection_id": 4294967295, "statement": "commit", "status": "success"}
2022/11/04 17:15:52.842687 +0800 INFO logtail/handle.go:399 [Logtail] record block meta row <0:272457-1023-1066-0-0>, true, 1667553118473704260-0, 1667553186354062457-0, 4dc0f265-216d-4235-a53d-842cf71b0644:11164_1984_1984:100:10909, 
2022/11/04 17:15:52.842747 +0800 INFO logtail/handle.go:399 [Logtail] record block meta row <0:272457-1023-1241-0-0>, false, 1667553186354062457-0, 0-0, 913916d0-19ee-4ac8-8398-35c7c334b541:10574_1600_1600:100:10319, 
2022/11/04 17:15:52.842802 +0800 INFO logtail/handle.go:471 [Logtail] send block meta for "bmsql_district"
2022/11/04 17:15:52.842818 +0800 INFO logtail/handle.go:471 [Logtail] send block meta for "bmsql_district"
2022/11/04 17:15:52.843196 +0800 INFO frontend/mysql_cmd_executor.go:3202 query trace {"connection_id": 1017, "query": "execute __mo_stmt_id_2 using @__mo_stmt_var_0,@__mo_stmt_var_1", "vars": "3 , 7"}
2022/11/04 17:15:52.843248 +0800 INFO frontend/mysql_cmd_executor.go:2702 time of Exec.Build : 1.968659ms
panic: runtime error: index out of range [11] with length 4

goroutine 209738 [running]:
github.com/matrixorigin/matrixone/pkg/container/batch.(*Batch).GetVector(...)
	/data1/sudong/mo-regression/matrixone/pkg/container/batch/batch.go:217
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*blockReader).Read(0xc00c388180, {0xc015775c80?, 0xc0062e4f00?, 0x416f1d?}, 0x222be80?, 0xc011931580?)
	/data1/sudong/mo-regression/matrixone/pkg/vm/engine/disttae/reader.go:57 +0x32d
github.com/matrixorigin/matrixone/pkg/vm/pipeline.(*Pipeline).Run(0xc0001eff28, {0x2906e18, 0xc00c388180}, 0xc00c382c00)
	/data1/sudong/mo-regression/matrixone/pkg/vm/pipeline/pipeline.go:68 +0x1aa
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).Run(0xc00342a180?, 0xc003440758?)
	/data1/sudong/mo-regression/matrixone/pkg/sql/compile/scope.go:55 +0xc8
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun.func1(0xc0001b9400?)
	/data1/sudong/mo-regression/matrixone/pkg/sql/compile/scope.go:73 +0x75
created by github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun
	/data1/sudong/mo-regression/matrixone/pkg/sql/compile/scope.go:68 +0x18d

**Comment by fengttt** [comment]
@aressu1985 It is a prepared statement.   Can we pull out the sql text of the prepared statement?

**Comment by fengttt** [comment]
@and schema.

**Comment by aressu1985** [comment]
UPDATE bmsql_district " +
   		"    SET d_ytd = d_ytd + ? " +
   		"    WHERE d_w_id = ? AND d_id = ?

**Comment by aressu1985** [comment]
drop database if exists tpcc;
create database if not exists tpcc;
use tpcc;

create table bmsql_config (
  cfg_name    varchar(30) primary key,
  cfg_value   varchar(50)
);

create table bmsql_warehouse (
  w_id        integer   not null,
  w_ytd       decimal(12,2),
  w_tax       decimal(4,4),
  w_name      varchar(10),
  w_street_1  varchar(20),
  w_street_2  varchar(20),
  w_city      varchar(20),
  w_state     char(2),
  w_zip       char(9),
  primary key (w_id)
) PARTITION BY KEY(w_id);

create table bmsql_district (
  d_w_id       integer       not null,
  d_id         integer       not null,
  d_ytd        decimal(12,2),
  d_tax        decimal(4,4),
  d_next_o_id  integer,
  d_name       varchar(10),
  d_street_1   varchar(20),
  d_street_2   varchar(20),
  d_city       varchar(20),
  d_state      char(2),
  d_zip        char(9),
  primary key (d_w_id, d_id)
) PARTITION BY KEY(d_w_id);

create table bmsql_customer (
  c_w_id         integer        not null,
  c_d_id         integer        not null,
  c_id           integer        not null,
  c_discount     decimal(4,4),
  c_credit       char(2),
  c_last         varchar(16),
  c_first        varchar(16),
  c_credit_lim   decimal(12,2),
  c_balance      decimal(12,2),
  c_ytd_payment  decimal(12,2),
  c_payment_cnt  integer,
  c_delivery_cnt integer,
  c_street_1     varchar(20),
  c_street_2     varchar(20),
  c_city         varchar(20),
  c_state        char(2),
  c_zip          char(9),
  c_phone        char(16),
  c_since        timestamp,
  c_middle       char(2),
  c_data         varchar(500),
  primary key (c_w_id, c_d_id, c_id)
) PARTITION BY KEY(c_w_id);

create table bmsql_history (
  hist_id  integer auto_increment,
  h_c_id   integer,
  h_c_d_id integer,
  h_c_w_id integer,
  h_d_id   integer,
  h_w_id   integer,
  h_date   timestamp,
  h_amount decimal(6,2),
  h_data   varchar(24),
  primary key (hist_id)
);

create table bmsql_new_order (
  no_w_id  integer   not null,
  no_d_id  integer   not null,
  no_o_id  integer   not null,
  primary key (no_w_id, no_d_id, no_o_id)
) PARTITION BY KEY(no_w_id);

create table bmsql_oorder (
  o_w_id       integer      not null,
  o_d_id       integer      not null,
  o_id         integer      not null,
  o_c_id       integer,
  o_carrier_id integer,
  o_ol_cnt     integer,
  o_all_local  integer,
  o_entry_d    timestamp,
  primary key (o_w_id, o_d_id, o_id)
) PARTITION BY KEY(o_w_id);

create table bmsql_order_line (
  ol_w_id         integer   not null,
  ol_d_id         integer   not null,
  ol_o_id         integer   not null,
  ol_number       integer   not null,
  ol_i_id         integer   not null,
  ol_delivery_d   timestamp,
  ol_amount       decimal(6,2),
  ol_supply_w_id  integer,
  ol_quantity     integer,
  ol_dist_info    char(24),
  primary key (ol_w_id, ol_d_id, ol_o_id, ol_number)
) PARTITION BY KEY(ol_w_id);

create table bmsql_item (
  i_id     integer      not null,
  i_name   varchar(24),
  i_price  decimal(5,2),
  i_data   varchar(50),
  i_im_id  integer,
  primary key (i_id)
) PARTITION BY KEY(i_id);

create table bmsql_stock (
  s_w_id       integer       not null,
  s_i_id       integer       not null,
  s_quantity   integer,
  s_ytd        integer,
  s_order_cnt  integer,
  s_remote_cnt integer,
  s_data       varchar(50),
  s_dist_01    char(24),
  s_dist_02    char(24),
  s_dist_03    char(24),
  s_dist_04    char(24),
  s_dist_05    char(24),
  s_dist_06    char(24),
  s_dist_07    char(24),
  s_dist_08    char(24),
  s_dist_09    char(24),
  s_dist_10    char(24),
  primary key (s_w_id, s_i_id)
) PARTITION BY KEY(s_w_id);

**Comment by fengttt** [comment]
@ouyuanning @aressu1985  can you check with tpcc run?

**Comment by ouyuanning** [comment]
> @ouyuanning @aressu1985 can you check with tpcc run?

sorry, i can't

**Comment by ouyuanning** [comment]
as my check in mo-tpcc. with

**Comment by aressu1985** [comment]
fixed:e23e00f1d01d43e33c509a6d239ccc38b5f93409
